components/patient-data/ui/src/main/resources/PhenoTips/JSONImportDialog.xml

```diff
@@ -0,0 +1,535 @@
+      jsonButton.observe("click", function (event) {
+        _this.openModal(event);
+      }.bindAsEventListener(this));
+
+      var buttons = new Element('div', {'class' : 'buttons import-block-bottom'});
+
+      buttons.insert(new Element('input',
+        {type: 'button',
+          name : 'import',
+          'value': "$escapetool.javascript($services.localization.render('phenotips.patientData.buttons.import'))",
+          'class' : 'button',
+          'id': 'import_button'
+        }).wrap('span', {'class' : 'buttonwrapper'}));
+
+      buttons.insert(new Element('input',
+        {type: 'button',
+          name : 'cancel',
+          'value': "$escapetool.javascript($services.localization.render('phenotips.patientData.buttons.cancel'))",
+          'class' : 'button secondary'})
+        .wrap('span', {'class' : 'buttonwrapper'}));
+
+      this.mainDiv.insert(buttons);
+
+      var cancelButton = buttons.down('input[name="cancel"]');
+      cancelButton.observe('click', function() {
+        _this.hide();
+      });
+
+      var importButton = buttons.down('input[name="import"]');
+      importButton.observe('click', function() {
+        _this._importJsonData();
+      });
+
+      this.textareaChangeFunction = function() {
+        importButton.disabled = _this.jsonData.value.trim() == "";
+      };
+
+      // different browsers require different event for onchange detection (special cases: IE, paste, backspace)
+      var eventList = ["input","propertychange", "mousemove", "keyup"];
+      eventList.each(function(item) {
+        _this.jsonData.observe(item, _this.textareaChangeFunction);
+      });
+    },
+
+    importServices : {
+      'single-patient-import' : {
+        'service' : "$xwiki.getURL('PhenoTips.ImportPatient', 'get')",
+        'contentType' : 'application/x-www-form-urlencoded',
+        'modalTitle' : "$escapetool.javascript($services.localization.render('phenotips.patientData.jsonImportDialog.singleImport.title'))",
+        'buildParam' : function(data, patientId) {
+          return {'id' : patientId, 'data' : data, 'outputSyntax' : 'plain'};
+        },
+        'callback' : function (event, importDialog) {
+          event.stop();
+          if (typeof event.findElement('.editbody') !== "undefined") {
+            // We are in edit mode.
+            var importAfterSave = function() {
+              document.stopObserving('xwiki:document:saved', importAfterSave);
+            };
+            importDialog.dialog.show();
+            document.observe('xwiki:document:saved', importAfterSave);
+            document.fire('xwiki:actions:save', {'continue' : true, 'form' : event.findElement('form')});
+          } else {
+            // View mode
+            importDialog.dialog.show();
+          }
+          importDialog.patientId = event.target.getAttribute("content");
+        }
+      },
+      'multi-patient-import' : {
+        'service' : 'rest/patients',
```

Comment:
This is fragile. Better use `XWiki.contextPath +  '/rest/patients'`, which is guaranteed to produce the correct URL regardless of the current path.